### PR TITLE
Backend feature/update order

### PR DESCRIPTION
**Pull request by burakberk**
- Implemented update order status functionality for vendor and customers.
- Implemented getting specific orders via mainOrderID and orderID.
- Implemented vendor order history access.
- Refactored shoppingCart, order and purchase to coding conventions of the repo.
-  Endpoints changed and now require less input from the frontend.

New endpoints relating shoppingCart, order:

*Vendor:
/vendor

get /order/main : get all orders
patch /order/main : change order status ** req body: mainOrderID, orderID

*Customer:
/customer

get /order/main : get all customers order
post /order/main : create order

get /order/orderID : get specific order ** req body: mainOrderID, orderID

patch /order/main : update order status ** req body: mainOrderID, orderID, status

post /shoppingCart/main : add to order to shoppingCart ** req body: productId, vendorId,  amount

get /shoppingCart/main : get shoppingCart

post /shoppingCart/delete : delete specific product from shoppingCart ** req body: productId, vendorId

post /shoppingCart/reset : reset shoppingCart

Also example request can be accessed from [this postman collection.](https://www.getpostman.com/collections/c68fa9c29df02562b827)

### CODE REVIEW

File: backend/routers/vendor.js [hunk 1]
```diff
@@ -75,4 +76,17 @@ router
   .patch(VendorController.updateMyProductRequestController, RequestHelper.returnResponse)
   .delete(VendorController.deleteMyProductRequestController, RequestHelper.returnResponse);
 
+router.get("/", VendorController.getAllVendorsController, RequestHelper.returnResponse);
+
+router
```
Comment (mtyazici) [inline backend/routers/vendor.js:79]: Why does the endpoints related to vendor seem to be added here. Is it related to the squash and merge problem?

Comment (burakberk) [reply backend/routers/vendor.js:79]: It seems so, I will remove the duplicate

**Review by oztasoi — APPROVED** [review]
Nice work, now our coding convention integration is 100%. 💯

**Comment by burakberk** [comment]
Thanks, merged with backend.

**Pull request merged**